HeterogeneousCore/CUDAUtilities/interface/GPUSimpleVector.h

```diff
@@ -83,6 +83,11 @@ template <class T> struct SimpleVector {
   __inline__ __host__ __device__ int capacity() const { return m_capacity; }
 
   __inline__ __host__ __device__ T *data() const { return m_data; }
+    
+  __inline__ __host__ __device__ void set_size(int size) { m_size = size; }
+    
+  __inline__ __host__ __device__ void set_data(T * data) { m_data = data; }
```

Comment:
I don't know if I like or not a `set_data()` method... other comments ?

Reply:
I don't particularly like. If kept it should be accompanied with `capacity` parameter.

In principle I'd expect the same be achieved with move/copy assignment. On the other hand, the pointed memory areas are copied CPU<->GPU outside of the `SimpleVector` so there could be cases where it is handier (with the current design of `SimpleVector` to use `set_data()`.

(I must say it took me a while to figure out what exactly happens in the code that uses the 'set_data()`. I'm afraid that having the memory owned outside of the `SimpleVector` and transferring the `SimpleVector` members and the data memory separately between CPU and GPU may lead to difficult-to-understand code. I'm fine with that now, but eventually we should aim to simplify.)

Reply:
Yes, I agree it should set also `capacity` (and reduce `size` if it exceeds the new `capacity`).

And I agree long term we should come up with a better approach (are get unified memory to work).

Reply:
On further thought, if one resets the `data` pointer, the object doesn't anymore know what would be the correct `size`, so it would have to be given from outside as well (e.g. if the `data` has valid elements already). To me this sounds like the job of move assignment (I came to the conclusion that copying should be forbidden as two copies would just have the same `data` pointers). But I can imagine that with the outsourced memory management also the results move operations may become difficult to follow.